cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -317,15 +317,20 @@ private Future<Void> schedule(PodRef podRef, long delay, TimeUnit unit) {
                 });
             } catch (Exception e) {
                 if (ctx.backOff.done()) {
-                    LOGGER.infoCr(reconciliation, "Could not roll pod {}, giving up after {} attempts. Total delay between attempts {}ms",
+                    LOGGER.infoCr(reconciliation, "Could not verify pod {} is up-to-date, giving up after {} attempts. Total delay between attempts {}ms",
                             podRef, ctx.backOff.maxAttempts(), ctx.backOff.totalDelayMs(), e);
                     ctx.promise.fail(e instanceof TimeoutException ?
                             new io.strimzi.operator.common.operator.resource.TimeoutException() :
                             e);
                 } else {
                     long delay1 = ctx.backOff.delayMs();
-                    LOGGER.infoCr(reconciliation, "Could not roll pod {} due to {}, retrying after at least {}ms",
-                            podRef, e, delay1);
+                    if (e instanceof ForceableProblem) {
+                        LOGGER.debugCr(reconciliation, "Will temporarily skip verifying pod {} is up-to-date due to {}, retrying after at least {}ms",
+                                podRef, e.getMessage(), delay1);
+                    } else {
+                        LOGGER.infoCr(reconciliation, "Will temporarily skip verifying pod {} is up-to-date due to {}, retrying after at least {}ms",
+                                podRef, e, delay1);
+                    }
```

Comment:
Does the `e` versus `e.getMessage()` really differ? I.e. do we need the if/else?

Reply:
For the case of `ForceableProblem`  

* We expect that this is a temp problem (i.e. it's a controller, so we're skipping it) that will be solved with retrying, so no need to print that on Info
* using `e` prints the exception class name which we already know, so there's no value of it, we only wanna print the message of the ForceableProblem

For the else
* It's an unknown exception, so we wanna show on INFO, and we wanna log the exception type, hence the e.

Reply:
Oh, I did not notice the different log levels. Why do you think `ForceableProblem` should not be on an info level? It will be retried but might provide useful information on why things are taking long. Especially since it sometimes takes a long time before it is forced.

Reply:
Makes sense. Check now.

I also renamed it to `ForeseeableProblem` which is the correct spelling of `Foreseeable` and overridden the `toString` to use just the class name + message (i.e. avoid the FQN or Outer$Inner styles). That way the message would look like

```
INFO: Will temporarily skip verifying pod my-kafka-cluster-0 is up-to-date due to ForeseeableProblem: Pod my-cluster-kafka-0 is currently the controller and there are other pods still to verify, retrying after at least 0 ms
```

Reply:
Was it supposed to be `Foreseeable`? I always thought it was `Forceable` as in we can force the restart despite this issue if needed. I guess @tombentley would know. See for example the comment here: https://github.com/strimzi/strimzi-kafka-operator/blob/e5c2189b64f06e668f81d570f852afc2dc6c842b/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java#L795 or the code snippet here: https://github.com/strimzi/strimzi-kafka-operator/blob/e5c2189b64f06e668f81d570f852afc2dc6c842b/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java#L391-L404 or here: https://github.com/strimzi/strimzi-kafka-operator/blob/e5c2189b64f06e668f81d570f852afc2dc6c842b/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java#L647-L655

Reply:
@scholzj oh yeah! that makes much more sense to me now! I was honestly thinking if @tombentley added this, then it can't be a this big of a typo! I should've asked first, my bad .

This [InfoQ article](https://www.infoq.com/articles/unsolved-problems-foreseeable/#:~:text=Foreseeable%20problems%20are%20those%20that,they%20are%20addressed%2C%20the%20better.) also gave me some reassurance it's Foreseeable  :D 

I'll revert this commit

Reply:
TBH, I'm seeing this exception in the code since it exists ... and my brain still wants to always read it and think about it as `Foreseeable` and not `Forceable`. I have to always think about it twice to correct myself. So you are not the first one to run into this :-D.

Reply:
Done